packages/msw-trpc/src/handler.ts

```diff
@@ -0,0 +1,285 @@
+import { http, HttpResponse, WebSocketLink, ws } from 'msw'
+import { Link } from './links.js'
+import { TRPCCombinedDataTransformer, TRPCError, getTRPCErrorFromUnknown } from '@trpc/server'
+import {
+  TRPC_ERROR_CODES_BY_KEY,
+  TRPC_ERROR_CODE_KEY,
+  defaultTransformer,
+  getHTTPStatusCodeFromError,
+} from '@trpc/server/unstable-core-do-not-import'
+import { Observable, Unsubscribable, observable } from '@trpc/server/observable'
+import { TRPCMswConfig } from './types.js'
+
+const getQueryInput = (req: Request, transformer: TRPCCombinedDataTransformer) => {
+  const inputString = new URL(req.url).searchParams.get('input')
+
+  if (inputString == null) return inputString
+
+  return transformer.input.deserialize(JSON.parse(inputString))
+}
+
+const getMutationInput = async (req: Request, transformer: TRPCCombinedDataTransformer) => {
+  const body = await req.json()
```

Comment:
With TRPC 11, it is possible that a request will contain FormData (ex: for file uploads), and this code will throw an error.

Reply:
Thanks for the input, as this code is not new I think we could put a new milestone to support different input types.

Reply:
https://github.com/maloguertin/msw-trpc/issues/43